ddl/ddl_worker.go

```diff
@@ -63,9 +63,11 @@ func (d *ddl) onDDLWorker() {
 			return
 		}
 
-		err := d.handleDDLJobQueue(&shouldCleanJobs)
+		isCleaned, err := d.handleDDLJobQueue(shouldCleanJobs)
 		if err != nil {
 			log.Errorf("[ddl] handle ddl job err %v", errors.ErrorStack(err))
+		} else {
+			shouldCleanJobs = !isCleaned
```

Comment:
It's possible that the `shouldCleanJobs` is false, but `isCleaned` return false, then we will clean job again.

Reply:
If err is nil, can we safely set shouldCleanJobs to false?
Then we can remove the bool return value.